index.d.ts

```diff
@@ -2,33 +2,29 @@ export { } from "./jsx";
 
 export as namespace skate;
 
-export interface HasProps<El extends Component> {
-  props: { [key: string]: skate.PropOptions<El, any>; };
-}
-
-export interface OnUpdatedCallback {
```

Comment:
I understand. You are right.
Base component class has each callbacks always. Not an "execute if it exists" model.
https://github.com/skatejs/skatejs/blob/316925b0b4a6d1d867abeafe8452123151a1b3d3/src/api/component.js#L291-L313